---
title: "CoNVaQ: A short introduction"
author: "Simon J. Larsen"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{CoNVaQ: A short introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Vignettes are long form documentation commonly included in packages. Because they are part of the distribution of the package, they need to be as compact as possible. The `html_vignette` output type provides a custom style sheet (and tweaks some options) to ensure that the resulting html is as small as possible. The `html_vignette` format:

- Never uses retina figures
- Has a smaller default figure size
- Uses a custom CSS stylesheet instead of the default Twitter Bootstrap style

## Input data format

CoNVaQ takes as input two sets of segmented CNV calls, one for each group of patients. Each set must be a `data.frame` with the following five columns in *in this order*:

1. `patient`: An identifier for the patient or sample the segment was identified in. All segments with equal patient identifiers will be considered from the same patient/sample.
2. `chr`: Chromosome the position is located in.
3. `start`: First genomic position of the segment in base pairs.
4. `end`: Last genomic position of the segment in base pairs.
5. `type`: The variant type of the segment. Must be one of "Gain", "Loss" or "LOH".

You can load the included example data set like this:

```{r}
data("example", package="rconvaq")
```

The `example` object contains two data frames, `disease` and `healthy`:

```{r}
head(example$disease)
head(example$healthy)
```

## Statistical model

```{r results="hide"}
library(rconvaq)
results <- convaq(
  example$disease, example$healthy,
  model = "statistical",
  name1 = "Disease", name2="Healthy",
  p.cutoff = 0.05
)
head(results)
```

```{r echo=FALSE}
knitr::kable(head(results))
```

Set the `qvalues` argument to `TRUE` to compute empirical p-values as well:

```{r results="hide"}
results <- convaq(
  example$disease, example$healthy,
  model = "statistical",
  name1 = "Disease", name2="Healthy",
  p.cutoff = 0.05,
  qvalues = TRUE,
  qvalues.rep = 2000
)
head(results[,1:7])
```

```{r echo=FALSE}
knitr::kable(head(results[,1:7]))
```

## Query model

## Vignette Info

Note the various macros within the `vignette` section of the metadata block above. These are required in order to instruct R how to build the vignette. Note that you should change the `title` field and the `\VignetteIndexEntry` to match the title of your vignette.

## Styles

The `html_vignette` template includes a basic CSS theme. To override this theme you can specify your own CSS in the document metadata as follows:

    output: 
      rmarkdown::html_vignette:
        css: mystyles.css

## Figures

The figure sizes have been customised so that you can easily put two images side-by-side. 

```{r, fig.show='hold'}
plot(1:10)
plot(10:1)
```

You can enable figure captions by `fig_caption: yes` in YAML:

    output:
      rmarkdown::html_vignette:
        fig_caption: yes

Then you can use the chunk option `fig.cap = "Your figure caption."` in **knitr**.

## More Examples

You can write math expressions, e.g. $Y = X\beta + \epsilon$, footnotes^[A footnote here.], and tables, e.g. using `knitr::kable()`.

```{r, echo=FALSE, results='asis'}
knitr::kable(head(mtcars, 10))
```

Also a quote using `>`:

> "He who gives up [code] safety for [code] speed deserves neither."
([via](https://twitter.com/hadleywickham/status/504368538874703872))
